python_modules/libraries/dagster-shared/dagster_shared/record/__init__.py

```diff
@@ -159,6 +159,8 @@ def _namedtuple_record_transform(
         _ORIGINAL_CLASS_FIELD: cls,
         _KW_ONLY_FIELD: kw_only,
         "__reduce__": _reduce,
+        "__eq__": _eq,
+        "__ne__": _ne,
```

Comment:
we need to make this an opt-in or do it on the decorated classes directly, its a big performance hit 

some context in https://github.com/dagster-io/dagster/pull/25240

I feel like it was stock on record at one point and i took it off but i cant find the blame

Reply:
or maybe just when fields is empty ? theres still risk of two objects with the same field layout returning a false positive so thats an argument for explicit flagging

Reply:
yeah I also share that memory of this being stock on record at some point, so it probably got removed for this reason. just gonna close this PR for now as I'm about to be gone for a couple weeks and I don't think this is particularly high pri